include/zephyr/bluetooth/gatt.h

```diff
@@ -1781,6 +1781,16 @@ enum {
 	 */
 	BT_GATT_SUBSCRIBE_FLAG_WRITE_PENDING,
 
+	/** @brief Sent flag
+	 *
+	 *  If set, indicates that a subscription request (CCC write) has
+	 *  already been sent in the active connection.
+	 *
+	 *  Used to avoid sending subscription requests multiple times when the
+	 *  @kconfig{CONFIG_BT_GATT_AUTO_RESUBSCRIBE} quirk is enabled.
+	 */
+	BT_GATT_SUBSCRIBE_FLAG_SENT,
```

Comment:
Both this and the `BT_GATT_SUBSCRIBE_FLAG_WRITE_PENDING` are internally used only, right? Whereas the `BT_GATT_SUBSCRIBE_FLAG_VOLATILE` and `BT_GATT_SUBSCRIBE_FLAG_NO_RESUB` flags are set by the user. 

Can't help but to wonder if they should be separated into internal and public flags, but that may be outside the scope of this PR, but perhaps we can at least add a comment here?

Reply:
Yes you're right. Is it ok if I don't add the comment now, but make the other PR tomorrow?

Reply:
I added the commit to this PR. I'm a bit unsure about it, on one end just renaming it like I did doesn't force the distinction, but adding another `ATOMIC_DEFINE` internal_flags would probably eat up one more word in memory for each subscribe params struct.
What do you think is best?

Reply:
Personally I would much rather add an internal atomic and use that (quite little) extra memory, than putting internally used values in the public API. 

I'd like to get input from @jhedberg and @alwa-nordic on this. 

If we create the internal atomic, then we could also, after this PR, start the process of moving the `BT_GATT_SUBSCRIBE_FLAG_WRITE_PENDING` flag too, as that has no business being public either IMO.